hazelcast/src/main/java/com/hazelcast/map/impl/operation/EntryOperator.java

```diff
@@ -0,0 +1,382 @@
+        this.partitionService = nodeEngine.getPartitionService();
+        EventService eventService = nodeEngine.getEventService();
+        this.hasEventRegistration = eventService.hasEventRegistration(SERVICE_NAME, mapName);
+        this.mapEventPublisher = mapServiceContext.getMapEventPublisher();
+        this.partitionId = recordStore.getPartitionId();
+        this.callerAddress = mapOperation.getCallerAddress();
+    }
+
+    private void setProcessor(Object processor) {
+        if (backup) {
+            backupProcessor = ((EntryBackupProcessor) processor);
+            entryProcessor = null;
+        } else {
+            entryProcessor = ((EntryProcessor) processor);
+            backupProcessor = null;
+        }
+    }
+
+    public static EntryOperator operator(MapOperation mapOperation) {
+        return new EntryOperator(null, null, false, mapOperation);
+    }
+
+    public static EntryOperator operator(Object processor, MapOperation mapOperation) {
+        return new EntryOperator(processor, null, false, mapOperation);
+    }
+
+    public static EntryOperator operator(Object processor, Predicate predicate, MapOperation mapOperation) {
+        return new EntryOperator(processor, predicate, false, mapOperation);
+    }
+
+    public static EntryOperator operator(Object processor, Predicate predicate,
+                                         boolean collectWanEvents, MapOperation mapOperation) {
+        return new EntryOperator(processor, predicate, collectWanEvents, mapOperation);
+    }
+
+    public EntryOperator init(Data dataKey, Object oldValue, Object newValue, Data result, EntryEventType eventType) {
+        this.dataKey = dataKey;
+        this.oldValue = oldValue;
+        this.newValue = newValue;
+        this.eventType = eventType;
+        this.result = result;
+        return this;
+    }
+
+    public EntryOperator operateOnKey(Data dataKey) {
+        init(dataKey, null, null, null, null);
+
+        if (belongsAnotherPartition(dataKey)) {
+            return this;
+        }
+
+        oldValue = recordStore.get(dataKey, backup);
+        Boolean locked = recordStore.isLocked(dataKey);
+
+        return operateOnKeyValueInternal(dataKey, clonedOrRawOldValue(), locked);
+    }
+
+    public EntryOperator operateOnKeyValue(Data dataKey, Object oldValue) {
+        return operateOnKeyValueInternal(dataKey, oldValue, null);
+    }
+
+    private EntryOperator operateOnKeyValueInternal(Data dataKey, Object oldValue, Boolean locked) {
+        init(dataKey, oldValue, null, null, null);
+
+        Map.Entry entry = createMapEntry(dataKey, oldValue, locked);
+        if (outOfPredicateScope(entry)) {
+            return this;
+        }
+
+        process(entry);
+        findModificationType(entry);
+        newValue = entry.getValue();
+
+        if (readOnly && eventType != null) {
+            throwModificationInReadOnlyException();
```

Comment:
could this `eventType != null` be a method with a good sounding name, like `entryWasModified` so that it's more readable?